### PR TITLE
Fix bug causing some Linux machines to freeze

### PR DESCRIPTION
**Pull request by JoshVorick**
I also fixed a small bug in hash rate calculation and made the kernel able to handle a larger global_item_size.

### CODE REVIEW

**Comment by JoshVorick** [comment]
Sorry for the ugly code. I'm working on making it neat without harming performance at all. Hopefully I'll have it done before you wake up and merge this.

**Comment by DavidVorick** [comment]
the freezing was caused by clock()? lol

File: gpu-miner.cl [hunk 1]
```diff
@@ -16,8 +16,12 @@ __kernel void nonceGrind(__global uchar *headerIn, __global uchar *hashOut, __gl
...
 	// Set nonce
 	private int id = get_global_id(0);
-	blockHeader[32] = id / 256;
-	blockHeader[33] = id % 256;
+	// Suppoer global work sizes of up to 256^5 - 1
+	blockHeader[32] = id / (256 * 256 * 256 * 256);
+	blockHeader[33] = id / (256 * 256 * 256);
+	blockHeader[34] = id / (256 * 256);
+	blockHeader[35] = id / 256;
+	blockHeader[36] = id % 256;
```
Comment (DavidVorick) [inline gpu-miner.cl]: at this point you might as well go to 8

**Comment by JoshVorick** [comment]
That's what Wolf` said. I guess this clock() fix worked for him.

**Pull request merged**